pkg/network/ebpf/c/protocols/helpers/pktbuf.h

```diff
@@ -93,6 +113,82 @@ static __always_inline __maybe_unused long pktbuf_load_bytes(pktbuf_t pkt, u32 o
     return 0;
 }
 
+static __always_inline __maybe_unused long pktbuf_load_bytes_from_current_offset(pktbuf_t pkt, void *to, u32 len)
+{
+    return pktbuf_load_bytes(pkt, pktbuf_data_offset(pkt), to, len);
+}
+
+typedef struct {
+    void *prog_array_map;
+    __u32 index;
+} pktbuf_tail_call_option_t;
+
+static __always_inline __maybe_unused long pktbuf_tail_call_compact(pktbuf_t pkt, pktbuf_tail_call_option_t *options)
+{
+    switch (pkt.type) {
+    case PKTBUF_SKB:
+        return bpf_tail_call_compat(pkt.skb, options[PKTBUF_SKB].prog_array_map, options[PKTBUF_SKB].index);
+    case PKTBUF_TLS:
+        return bpf_tail_call_compat(pkt.ctx, options[PKTBUF_TLS].prog_array_map, options[PKTBUF_TLS].index);
+    }
+
+    pktbuf_invalid_operation();
+    return 0;
+}
+
+typedef struct {
+    void *map;
+    void *key;
+} pktbuf_map_lookup_option_t;
+
+static __always_inline __maybe_unused void* pktbuf_map_lookup(pktbuf_t pkt, pktbuf_map_lookup_option_t *options)
+{
+    switch (pkt.type) {
```

Comment:
Maybe just `return bpf_map_lookup_elem(options[pkt.type].map, options[pkt.type].key)`? Similarly for the other ones.